### PR TITLE
#170070968 create the signup endpoint v2 feature

### PR DESCRIPTION
**Pull request by YankyJayChris**
> #### What does this PR do?
> PR for create the signup endpoint v2 feature

> #### Description of Task to be completed
> - create user model
> - create hash password helper
> - create jwt token helper
> - signup endpoint
> - create signup validation

> #### How should this be manually tested?
> download the repo then run npm run serve then go to localhost:5000/api/v2/auth/signup
then provide required field
 1. firstname
 2. lastname
3. username
4. email
5. phoneNumber
6. password
7. re_password
> ### optional:
 > - avatar(jpeg,pjpeg,png)

> #### What are the relevant pivotal tracker stories?
> https://www.pivotaltracker.com/story/show/170070968
> #### Screenshots
> N/A

### CODE REVIEW

File: server/tests/v2/signin.test.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+//     phoneNumber: '0721314151',
+//     password: 'Password12',
+//     re_password: 'Password12',
+//   };
+//   before(async () => {
+//     try {
+//       const res = await chai.request(server)
+//         .post('/api/v2/auth/signup')
+//         .send(user);
+//       console.log(res.status);
```
Comment (mcaleb808) [inline server/tests/v2/signin.test.js]: @YankyJayChris Kindly avoid pushing commented code

File: server/v2/controllers/Users.js [hunk 2]
```diff
@@ -63,8 +65,8 @@ const User = {
     @param {object} res
     @returns {object} users array
    */
-  signin(req, res) {
-    const user = userModel.findOneByEmail(req.body.email);
+  async signin(req, res) {
+    const user = await userModel.findOneByEmail(req.body.email);
```
Comment (mcaleb808) [inline server/v2/controllers/Users.js]: Are you done with this `signin` controller?

File: server/v2/models/User.js [hunk 3]
```diff
@@ -1,84 +1,171 @@
...
+        firstname VARCHAR(128) NOT NULL,
+        lastname VARCHAR(128) NOT NULL,
+        email VARCHAR(128) NOT NULL,
+        username VARCHAR(128) NOT NULL,
+        type VARCHAR(128) NOT NULL,
+        phoneNumber VARCHAR(128) NOT NULL,
+        avatar VARCHAR(128) NOT NULL,
+        password VARCHAR(128) NOT NULL,
+        createdDate TIMESTAMP,
+        modifiedDate TIMESTAMP
```
Comment (mcaleb808) [inline server/v2/models/User.js]: Phone number and username should also be unique

File: server/v2/routes/Auth.js [hunk 4]
```diff
@@ -1,11 +1,18 @@
...
+// redflag middleware
+import userValidator from '../validation/User';
+import { uploadsingle } from '../helpers/upload';
+// import jwtToken from '../helpers/jwtToken';
 
 const router = express.Router();
 
 router.post('/test', (req, res) => res.status(200).json({ msg: 'it\'s works' }));
+router.post('/signup', uploadsingle, userValidator('signup', 'body'), Users.register);
+// router.post('/signin', userValidator('usersignin', 'body'), Users.signin);
```
Comment (mcaleb808) [inline server/v2/routes/Auth.js]: Kindly avoid pushing commented code

File: server/v2/controllers/Users.js (hunk 2 above)
Comment (YankyJayChris) [reply server/v2/controllers/Users.js]: yes am done

**Pull request merged**